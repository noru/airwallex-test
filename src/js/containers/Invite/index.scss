.page-invite {
  height: 100vh;
  width: 100%;
  display: flex;
  flex-flow: column;
  justify-content: space-between;

  .invite-header {
    font-size: 1.5em;
    padding: 1em 3em;
    border-bottom: 1px solid gray;
  }

  .invite-footer {
    text-align: center;
    border-top: 1px solid gray;
    padding: 2em;
  }

  .invite-content {
    text-align: center;
    .main-title {
      font-size: 3em;
      width: 50%;
      display: inline-block;
    }
    @media (max-width: 600px) {
      .main-title {
        font-size: 1.7em;
      }
    }
    .sub-title {
      padding: 2em 0;
    }
  }

  .invite-form-wrapper {
    display: flex;
    justify-content: center;

    .invite-form {
      width: 75%;
      background: white;
      padding: 1.5em;

      > * {
        margin-bottom: 1.5em;
      }

      .form-title {
        text-align: center;

        &:after {
          content: '_____';
          display: block;
          font-weight: bold;
        }
      }

      .button {
        width: 100%;
      }
    }
  }
}