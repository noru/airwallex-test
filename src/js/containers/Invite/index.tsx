import React from 'react'
import { translate } from 'react-i18next'
import { connect } from 'dva'
import { flow } from 'noru-utils/lib'

export class Login extends React.Component<any, any> {

  click = () => {
    let { dispatch } = this.props
    dispatch({ type: 'invite/get/invite-response'})
  }

  render() {
    return (
      <div onClick={this.click}>
        stub
        <span>{JSON.stringify(this.props.inviteResponse)}</span>
      </div>
    )
  }
}

export default flow(
  Login,
  translate(),
  connect(({ invite: { inviteResponse }}) => ({ inviteResponse })),
)