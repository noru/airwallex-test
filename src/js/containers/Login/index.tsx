import React from 'react'
import { translate } from 'react-i18next'

export class Login extends React.Component<any, any> {

  render() {
    return (
      <div>
        stub
      </div>
    )
  }
}

export default translate()(Login)