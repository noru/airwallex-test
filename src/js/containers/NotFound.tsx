import * as React from 'react'

export default class NotFound extends React.PureComponent<any, any> {

  render() {

    return (
      <div className="not-found-page" >
        <span>404. You know what to do.</span>
      </div>
    )
  }

}
