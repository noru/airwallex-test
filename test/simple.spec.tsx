
import React from 'react'
import { Invite } from '../src/js/containers/Invite'
// import { identity } from 'noru-utils/lib'
import renderer from 'react-test-renderer'

const identity = i => i

let dispatchCallCount = 0
const dispatchMock = () => dispatchCallCount++

test('General tests', () => {
  let component = renderer.create(
    <Invite t={identity} dispatch={dispatchMock}></Invite>,
  )
  let instance = component.getInstance()
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  instance.openModal()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  instance.done()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(dispatchCallCount).toBe(1)

  instance.invite()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(dispatchCallCount).toBe(3)

})