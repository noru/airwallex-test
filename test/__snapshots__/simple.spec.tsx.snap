// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General tests 1`] = `
<div
  className="page-invite"
>
  <div
    className="invite-header"
  >
    Broccoli & Co.
  </div>
  <div
    className="invite-content"
  >
    <div
      className="main-title"
    >
      A better way to enjoy everday.
    </div>
    <div
      className="sub-title"
    >
      Be the first to know when we launch.
    </div>
    <a
      className="button"
      onClick={[Function]}
    >
      Request an invite
    </a>
  </div>
  <div
    className="invite-footer"
  >
    <div>
      Made with ❤️ in Melbourne
    </div>
    <div>
      © 2016 Broccoli & Co. All rights reserved.
    </div>
  </div>
</div>
`;

exports[`General tests 2`] = `
<div
  className="page-invite"
>
  <div
    className="invite-header"
  >
    Broccoli & Co.
  </div>
  <div
    className="invite-content"
  >
    <div
      className="main-title"
    >
      A better way to enjoy everday.
    </div>
    <div
      className="sub-title"
    >
      Be the first to know when we launch.
    </div>
    <a
      className="button"
      onClick={[Function]}
    >
      Request an invite
    </a>
  </div>
  <div
    className="invite-footer"
  >
    <div>
      Made with ❤️ in Melbourne
    </div>
    <div>
      © 2016 Broccoli & Co. All rights reserved.
    </div>
  </div>
  <div
    className="modal is-active"
  >
    <div
      className="modal-background"
      onClick={[Function]}
    />
    <div
      className="modal-content"
    >
      <div
        className="invite-form-wrapper"
      >
        <div
          className="invite-form"
        >
          <form
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="form-title"
            >
              Request an invite
            </div>
            <input
              className="input"
              disabled={false}
              name="local.fullname"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              placeholder={
                Array [
                  "Full name",
                ]
              }
              required={true}
              type="text"
              value=""
            />
            <input
              className="input"
              disabled={false}
              name="local.email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              placeholder={
                Array [
                  "Email",
                ]
              }
              required={true}
              type="email"
              value=""
            />
            <input
              className="input"
              disabled={false}
              name="local.confirm"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              placeholder={
                Array [
                  "Confirm Email",
                ]
              }
              required={true}
              type="email"
              value=""
            />
            <button
              className="button is-primary"
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`General tests 3`] = `
<div
  className="page-invite"
>
  <div
    className="invite-header"
  >
    Broccoli & Co.
  </div>
  <div
    className="invite-content"
  >
    <div
      className="main-title"
    >
      A better way to enjoy everday.
    </div>
    <div
      className="sub-title"
    >
      Be the first to know when we launch.
    </div>
    <a
      className="button"
      onClick={[Function]}
    >
      Request an invite
    </a>
  </div>
  <div
    className="invite-footer"
  >
    <div>
      Made with ❤️ in Melbourne
    </div>
    <div>
      © 2016 Broccoli & Co. All rights reserved.
    </div>
  </div>
</div>
`;

exports[`General tests 4`] = `
<div
  className="page-invite"
>
  <div
    className="invite-header"
  >
    Broccoli & Co.
  </div>
  <div
    className="invite-content"
  >
    <div
      className="main-title"
    >
      A better way to enjoy everday.
    </div>
    <div
      className="sub-title"
    >
      Be the first to know when we launch.
    </div>
    <a
      className="button"
      onClick={[Function]}
    >
      Request an invite
    </a>
  </div>
  <div
    className="invite-footer"
  >
    <div>
      Made with ❤️ in Melbourne
    </div>
    <div>
      © 2016 Broccoli & Co. All rights reserved.
    </div>
  </div>
</div>
`;
